### ISSUE TITLE
Apple Silicon (ARM) No native build was found for platform=linux arch=arm64

### ISSUE THREAD

**Issue by danielefiungo** [nrwl/nx#13973]
### Current Behavior

On Apple Silicon (ARM) Mac:

- vscode + NX + devcontainer isn't working. 
- Installing NX globally inside the container  (npm i -g nx) don't work

### Expected Behavior

Have a native build for platform=linux arch=arm64, allow NX extension to work in dev container

### Steps to Reproduce

1.  inside a arm linux container (vscode devcontainer)
2. issue any NX commad

### Nx Report

```shell
>  NX   No native build was found for platform=linux arch=arm64 runtime=node abi=108 uv=1 armv=8 libc=glibc node=18.12.1

       loaded from: /workspaces/ic/node_modules/@parcel/watcher
   
   
   Because of the error the Nx daemon process has exited. The next Nx command is going to restart the daemon process.
   If the error persists, please run "nx reset".
```

### Failure Logs

```shell
/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:60
  throw new Error('No native build was found for ' + target + '\n    loaded from: ' + dir + '\n')
  ^

Error: No native build was found for platform=linux arch=arm64 runtime=electron abi=93 uv=1 armv=8 libc=glibc node=16.14.2
    loaded from: /home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/@parcel/watcher

    at Function.load.path (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:60:9)
    at load (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:22:30)
    at Object.<anonymous> (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/@parcel/watcher/index.js:1:42)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/nxls/main.js:213:9822)
[Info  - 6:57:03 AM] Connection to server got closed. Server will restart.
[Error - 6:57:03 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 6:57:03 AM] Nx Console Client client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 6:57:03 AM] Sending request nx/workspace failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 6:57:03 AM] Sending request nx/workspace failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:60
  throw new Error('No native build was found for ' + target + '\n    loaded from: ' + dir + '\n')
  ^

Error: No native build was found for platform=linux arch=arm64 runtime=electron abi=93 uv=1 armv=8 libc=glibc node=16.14.2
    loaded from: /home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/@parcel/watcher

    at Function.load.path (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:60:9)
    at load (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:22:30)
    at Object.<anonymous> (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/@parcel/watcher/index.js:1:42)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/nxls/main.js:213:9822)
[Info  - 6:57:03 AM] Connection to server got closed. Server will restart.
[Error - 6:57:03 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 6:57:03 AM] Nx Console Client client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:60
  throw new Error('No native build was found for ' + target + '\n    loaded from: ' + dir + '\n')
  ^

Error: No native build was found for platform=linux arch=arm64 runtime=electron abi=93 uv=1 armv=8 libc=glibc node=16.14.2
    loaded from: /home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/@parcel/watcher

    at Function.load.path (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:60:9)
    at load (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:22:30)
    at Object.<anonymous> (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/@parcel/watcher/index.js:1:42)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/nxls/main.js:213:9822)
[Info  - 6:57:03 AM] Connection to server got closed. Server will restart.
[Error - 6:57:03 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 6:57:03 AM] Nx Console Client client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 6:57:03 AM] Restarting server failed
  Message: Pending response rejected since connection got disposed
  Code: -32097 
/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:60
  throw new Error('No native build was found for ' + target + '\n    loaded from: ' + dir + '\n')
  ^

Error: No native build was found for platform=linux arch=arm64 runtime=electron abi=93 uv=1 armv=8 libc=glibc node=16.14.2
    loaded from: /home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/@parcel/watcher

    at Function.load.path (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:60:9)
    at load (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:22:30)
    at Object.<anonymous> (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/@parcel/watcher/index.js:1:42)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/nxls/main.js:213:9822)
[Info  - 6:57:03 AM] Connection to server got closed. Server will restart.
[Error - 6:57:03 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 6:57:03 AM] Nx Console Client client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 6:57:03 AM] Restarting server failed
  Message: Pending response rejected since connection got disposed
  Code: -32097 
/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:60
  throw new Error('No native build was found for ' + target + '\n    loaded from: ' + dir + '\n')
  ^

Error: No native build was found for platform=linux arch=arm64 runtime=electron abi=93 uv=1 armv=8 libc=glibc node=16.14.2
    loaded from: /home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/@parcel/watcher

    at Function.load.path (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:60:9)
    at load (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/node-gyp-build/index.js:22:30)
    at Object.<anonymous> (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/node_modules/@parcel/watcher/index.js:1:42)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/vscode/.vscode-server/extensions/nrwl.angular-console-17.27.1/nxls/main.js:213:9822)
[Error - 6:57:03 AM] The Nx Console Client server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
[Error - 6:57:03 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 6:57:03 AM] Nx Console Client client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 6:57:03 AM] Restarting server failed
  Message: Pending response rejected since connection got disposed
  Code: -32097
```

**Comment by AgentEnder** [comment]
Hey, this looks like an issue with @parcel/watcher. Can you open an issue on their repo? https://github.com/parcel-bundler/watcher

**Comment by simpros** [comment]
This looks like really dead on their side...

**Comment by sampullman** [comment]
`@parcel/watcher` added pre-compiled binaries for arm in a [recent release](https://github.com/parcel-bundler/watcher/releases), which should make it possible to fix this issue by bumping the dependency in `@nx/workspace` to `2.2.0`

**Comment by Cammisuli** [comment]
fixed by #19751
